### PR TITLE
Mes 9002 | vehicle checks modal validation

### PR DESCRIPTION
**Pull request by gittins7**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/waiting-room-to-car/components/vehicle-checks-question/vehicle-checks-question.ts [hunk 1]
```diff
@@ -94,4 +122,21 @@ export class VehicleChecksQuestionComponent implements OnChanges {
...
+
+  displayErrorMessage(): boolean {
+    return this.submitClicked && this.questionFormControl.valid && !this.questionOutcomeFormControl.valid;
+  }
+
+  setInvalidIfUndefined(control: UntypedFormControl) {
+    if (control.value === undefined) {
+      control.setErrors({ 'invalid': true });
+    }
+  }
```
Comment (matthew2564) [inline src/app/pages/waiting-room-to-car/components/vehicle-checks-question/vehicle-checks-question.ts]: Do we need this? Can we not use standard form control way and mark both controls as `[Validators.required]`?

File: src/app/pages/waiting-room-to-car/components/vehicle-checks/vehicle-checks.ts [hunk 2]
```diff
@@ -91,10 +97,15 @@ export class VehicleChecksComponent implements OnChanges {
...
-      componentProps: { category: this.category },
+      componentProps: {
+        category: this.category,
+        modalOpened: this.onOpenVehicleChecksModal,
+        submitClicked: this.submitClicked,
+      },
       cssClass: `modal-fullscreen ${this.accessibilityService.getTextZoomClass()}`,
     });
     await modal.present();
+    this.onOpenVehicleChecksModal.emit();
```
Comment (matthew2564) [inline src/app/pages/waiting-room-to-car/components/vehicle-checks/vehicle-checks.ts]: Where is this being listened out for? Is it still needed?

File: src/app/pages/waiting-room-to-car/components/vehicle-checks-question/vehicle-checks-question.ts [hunk 3]
```diff
@@ -37,11 +47,26 @@ export class VehicleChecksQuestionComponent implements OnChanges {
...
+      this.setInvalidIfUndefined(this.questionOutcomeFormControl);
+    });
+
+    this.ref.detectChanges();
+  }
+
+  ngOnDestroy() {
+    this.onDestroy$.next();
+    this.onDestroy$.complete();
+  }
```
Comment (matthew2564) [inline src/app/pages/waiting-room-to-car/components/vehicle-checks-question/vehicle-checks-question.ts]: Why is this needed?

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]
